### PR TITLE
Moka Test Interview

### PR DESCRIPTION
**Pull request by willymiiee**
Hi, this is my code for Moka Interview Test.
I hope my code is acceptable.
Thanks.

### CODE REVIEW

**Comment by glaksmono** [comment]
Thanks for creating the PR @willymiiee 

I have a question. Why do you think this could happened?
```
mokatest=# select * from businesses;
    id     |  name   |     address      |  city   |  user_id  |          created_at           |          updated_at
-----------+---------+------------------+---------+-----------+-------------------------------+-------------------------------
 980190962 |         |                  |         |           | 2017-01-13 20:16:24+07        | 2017-01-13 20:16:24+07
 298486374 |         |                  |         |           | 2017-01-13 20:16:24+07        | 2017-01-13 20:16:24+07
 980190971 | GL Cafe | JL Gatot Subroto | Jakarta | 980190978 | 2017-01-13 20:17:29.254075+07 | 2017-01-13 20:17:29.254075+07
(3 rows)

mokatest=# select * from items;
    id     | name | price | business_id |       created_at       |       updated_at
-----------+------+-------+-------------+------------------------+------------------------
 980190962 |      |       |             | 2017-01-13 20:16:24+07 | 2017-01-13 20:16:24+07
 298486374 |      |       |             | 2017-01-13 20:16:24+07 | 2017-01-13 20:16:24+07
(2 rows)

mokatest=#
```

Reference: https://github.com/willymiiee/moka-rails-interview/blob/master/db/seeds.rb

File: app/controllers/businesses_controller.rb [hunk 1]
```diff
@@ -1,3 +1,49 @@
 class BusinessesController < ApplicationController
+  before_action :set_business, only: [:show, :edit, :update, :destroy]
```
Comment (glaksmono) [inline app/controllers/businesses_controller.rb]: I believe you never use `destroy` here - applies everywhere else

File: test/controllers/users_controller_test.rb [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+  setup do
+    @user = {
+   		first_name: "W",
+   		last_name:"N", 
+   		email: "w@n.com",
+   		password:"12345678",
+   		password_confirmation:"12345678"
+    }
+
+    @userUpdate = {
```
Comment (glaksmono) [inline test/controllers/users_controller_test.rb]: `camelCase`, while everywhere else is `snake`

File: test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -1,7 +1,58 @@
...
+      first_name: "A",
+      last_name: "B",
+      email: "a@b.com",
+      password: "123456",
+      password_confirmation: "123456"
+    }
+  end
+
+ 	def create_user
```
Comment (glaksmono) [inline test/controllers/users_controller_test.rb]: indentation

File: test/controllers/items_controller_test.rb [hunk 4]
```diff
@@ -1,7 +1,93 @@
...
+  test "should destroy item" do
+    login
+    create_business
+    create_item
+    assert_difference('Item.count', -1) do
+      delete :destroy, id: @item
+    end
+
+    assert_redirected_to items_path
+  end
```
Comment (glaksmono) [inline test/controllers/items_controller_test.rb:108]: no "should fail" scenarios for controller tests

File: app/controllers/items_controller.rb [hunk 5]
```diff
@@ -10,15 +12,53 @@ def index
...
+  # GET /items/1/edit
+  def edit
+    :set_item
+  end
+
+  # POST /items
   def create
-    # TODO: [MOKA-001] Finish the implementation of Create/Update/Delete for Items
+    @item = Item.new(item_params)
+    @item.price = item_params[:price].gsub(/\D/, '').to_i
```
Comment (glaksmono) [inline app/controllers/items_controller.rb]: pretty smart here 👍

File: app/models/user.rb [hunk 6]
```diff
@@ -1,10 +1,14 @@
 class User < ActiveRecord::Base
 
+  before_save { self.email = email.downcase }
```
Comment (glaksmono) [inline app/models/user.rb:3]: 👍

**Comment by willymiiee** [comment]
Hi Mr. Laksmono,

I've added some code and logic, I hope my code is acceptable :smile: 

From your previous question, I forgot to change the fixtures on `test/fixtures`, so it will insert an empty object.

**Comment by glaksmono** [comment]
Great that you change the fixtures, but it will still replace the `seeds.rb` data though :)

```
mokatest=# select * from users;
-[ RECORD 1 ]---+-------------------------------------------------------------
id              | 1
email           | john@moka.com
password_digest | $2a$10$X2Hg2kxQAJuUpgfJd/G0tueld4/mKQxySKC20ax0ZIlHj6/EdesKy
first_name      | John
last_name       | Doe
created_at      | 2017-01-17 09:56:20.56955+07
updated_at      | 2017-01-17 09:56:20.56955+07
-[ RECORD 2 ]---+-------------------------------------------------------------
id              | 2
email           | jane@moka.com
password_digest | $2a$10$RBWnMtVcTfCwML1WJD68Ie6BjyOm.40EFl.aunObVYFzZMRkcixJy
first_name      | Jane
last_name       | Doe
created_at      | 2017-01-17 09:56:20.640416+07
updated_at      | 2017-01-17 09:56:20.640416+07

mokatest=# select * from users;
-[ RECORD 1 ]---+-----------------------
id              | 980190962
email           | arnold.rambo@one
password_digest |
first_name      | Arnold
last_name       | Rambo
created_at      | 2017-01-17 09:56:36+07
updated_at      | 2017-01-17 09:56:36+07
-[ RECORD 2 ]---+-----------------------
id              | 298486374
email           | rocky.predator@two
password_digest |
first_name      | Rocky
last_name       | Predator
created_at      | 2017-01-17 09:56:36+07
updated_at      | 2017-01-17 09:56:36+07

mokatest=#
```

May want to take a look at this? http://stackoverflow.com/questions/1898782/prevent-rails-test-from-deleting-seed-data

Can you fix this?

**Comment by willymiiee** [comment]
Hi Mr. Laksmono,

I've commented the load fixtures part and loaded the `seeds.rb` file on `test_helper.rb`.

Thanks :grin:

**Comment by glaksmono** [comment]
👍

**Pull request closed without merging**